Bump Turing v0.40

Red-Portal opened the pull request.

Red-Portal committed:
bump Turing v0.40

Red-Portal commented:
Hi @penelopeysm , could you take a quick look to make sure everything is as intended upstream?

penelopeysm commented on ext/SliceSamplingTuringExt.jl:
```diff
@@ -32,19 +32,13 @@ const SliceSamplingStates = Union{
     vi     = Turing.DynamicPPL.VarInfo(rng, model, Turing.SampleFromUniform())
-    vi_spl = last(Turing.DynamicPPL.evaluate!!(model, rng, vi, Turing.SampleFromUniform()))
-    θ      = vi_spl[:]
+    vi_spl = last(Turing.DynamicPPL.evaluate_and_sample!!(rng, model, vi, Turing.SampleFromUniform()))
```
Hmmm this will only work on DynamicPPL 0.37, so only Turing 0.40, so I think it might be better to restrict the compat to only 0.40.

Alternatively to be explicit you could make both Turing and DynamicPPL triggers for the Turing extension, then that way you can control both compat entries. I think that's the best way although it is a bit of boilerplate.

Otherwise LGTM :)

Red-Portal replied:
Ah okay. Hmm let me just restrict to Turing 0.40 to keep things simple. Thanks for taking a look!

Red-Portal committed:
fix tighten compat for Turing

`Turing.DynamicPPL.evaluate_and_sample!!` is only available on
DynamicPPL v0.37, which is re-exported only on Turing
v0.40.

Red-Portal committed:
fix formatting

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Red-Portal committed:
fix add tests for drawing initial parameter

Red-Portal committed:
increment path version

Red-Portal committed:
run formatter

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
